packages/openid4vc-holder/src/presentations/PresentationExchangeService.ts

```diff
@@ -0,0 +1,360 @@
+      })
+    })
+
+    for (const [subjectId, inputDescriptors] of Object.entries(vps)) {
+      // Determine a suitable verification method for the presentation
+      const verificationMethod = await this.getVerificationMethodForSubjectId(agentContext, subjectId)
+
+      if (!verificationMethod) {
+        throw new AriesFrameworkError(`No verification method found for subject id ${subjectId}`)
+      }
+
+      const inputDescriptorsForVp = (presentationDefinition.input_descriptors as PresentationDefinitionV1[]).filter(
+        (inputDescriptor) => inputDescriptor.id in inputDescriptors
+      )
+
+      const credentialsForVp = Object.values(inputDescriptors)
+        .flatMap((inputDescriptors) => inputDescriptors)
+        .map(getSphereonW3cVerifiableCredential)
+
+      const presentationDefinitionForVp = {
+        ...presentationDefinition,
+        input_descriptors: inputDescriptorsForVp,
+
+        // We remove the submission requirements, as it will otherwise fail to create the VP
+        // FIXME: Will this cause issue for creating the credential? Need to run tests
+        submission_requirements: undefined,
+      }
+
+      // Q1: is holder always subject id, what if there are multiple subjects???
+      // Q2: What about proofType, proofPurpose verification method for multiple subjects?
+      const verifiablePresentationResult = await this.pex.verifiablePresentationFrom(
+        presentationDefinitionForVp,
+        credentialsForVp,
+        this.getPresentationSignCallback(
+          agentContext,
+          verificationMethod,
+          // Can't include submission if more than one VP
+          Object.values(vps).length > 1 ? false : includePresentationSubmissionInVp
+        ),
+        {
+          holderDID: subjectId,
+          proofOptions: {
+            challenge,
+            domain,
+            nonce,
+          },
+          signatureOptions: {
+            verificationMethod: verificationMethod?.id,
+          },
+        }
+      )
+
+      verifiablePresentationResults.push(verifiablePresentationResult)
+    }
+
+    const firstVerifiablePresentationResult = verifiablePresentationResults[0]
+    if (!firstVerifiablePresentationResult) {
+      throw new AriesFrameworkError('No verifiable presentations created.')
+    }
+
+    const presentationSubmission: PexPresentationSubmission = {
+      id: firstVerifiablePresentationResult.presentationSubmission.id,
+      definition_id: firstVerifiablePresentationResult.presentationSubmission.definition_id,
+      descriptor_map: [],
+    }
+
+    for (const vp of verifiablePresentationResults) {
+      presentationSubmission.descriptor_map.push(
+        ...vp.presentationSubmission.descriptor_map.map((descriptor): Descriptor => {
+          const index = verifiablePresentationResults.indexOf(vp)
+          const prefix = verifiablePresentationResults.length > 1 ? `$[${index}]` : '$'
+          return {
+            format: 'jwt_vp',
```

Comment:
Would be good to use an enum for this format and the one below in the `path_nested` object.